src/nix/flake.cc

```diff
@@ -1303,8 +1303,10 @@ struct CmdFlakeShow : FlakeCommand, MixJSON
                         try {
                             if (visitor.isDerivation())
                                 showDerivation();
-                            else
-                                throw Error("expected a derivation");
+                            else {
+                                auto name = visitor.getAttrPathStr(state->s.name);
+                                logger->warn(fmt("%s must be a derivation", name));
```

Comment:
I think because `nix flake show`'s primary purpose isn't checking but describing, I think this phrasing is slightly better:

```suggestion
                                logger->warn(fmt("%s is not a derivation", name));
```